src/DynamoCoreWpf/Controls/InPorts.xaml.cs

```diff
@@ -0,0 +1,517 @@
+                Background = Brushes.Transparent
+            };
+
+            Grid.SetColumn(NodeAutoCompleteHover, 0);
+
+            nodeAutoCompleteMarker = new Border
+            {
+                Name = "NodeAutoCompleteMarker",
+                Cursor = Cursors.Hand,
+                CornerRadius = new CornerRadius(10),
+                Height = 20,
+                Width = 20,
+                VerticalAlignment = VerticalAlignment.Center,
+                HorizontalAlignment = HorizontalAlignment.Left,
+                Background = _nodeTransientOverlayColor,
+                SnapsToDevicePixels = true,
+                Visibility = Visibility.Collapsed
+            };
+
+            var nodeAutoCompleteMarkerLabel = new Label
+            {
+                Name = "NodeAutoCompleteMarkerLabel",
+                FontSize = 12,
+                Width = 25,
+                Height = 25,
+                Margin = new Thickness(-3, -3, 0, 0),
+                Content = "✨"
+            };
+
+            nodeAutoCompleteMarker.Child = nodeAutoCompleteMarkerLabel;
+            NodeAutoCompleteHover.Children.Add(nodeAutoCompleteMarker);
+
+            MainGrid.Children.Add(PortSnapping);
+            MainGrid.Children.Add(PortBackgroundBorder);
+            MainGrid.Children.Add(PortValueMarker);
+            MainGrid.Children.Add(PortDefaultValueMarker);
+            MainGrid.Children.Add(PortNameTextBox);
+            MainGrid.Children.Add(NodeAutoCompleteHover);
+
+            this.Content = MainGrid;
+
+            DataContextChanged += OnDataContextChanged;
+            Unloaded += OnPortViewUnloaded;
+        }
+
+        private void OnDataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
+        {
+            if (null != viewModel) return;
+
+            viewModel = e.NewValue as InPortViewModel;
+            viewModel.PropertyChanged += OnPropertyChanged;
+
+            //Initialize the PortBackgroundColor
+            PortBackgroundBorder.Background = viewModel.PortBackgroundColor;
+
+            var mouseLeftButtonDownTrigger = new Dynamo.UI.Views.HandlingEventTrigger()
+            {
+                EventName = "MouseLeftButtonDown",
+            };
+            var mouseLeftButtonDownAction = new InvokeCommandAction()
+            {
+                Command = viewModel.ConnectCommand,
+            };
+
+            mouseLeftButtonDownTrigger.Actions.Add(mouseLeftButtonDownAction);
+            Dynamo.Microsoft.Xaml.Behaviors.Interaction.GetTriggers(MainGrid).Add(mouseLeftButtonDownTrigger);
+
+            var mouseRightButtonDownTrigger = new Dynamo.UI.Views.HandlingEventTrigger()
+            {
+                EventName = "MouseRightButtonDown",
+            };
+            var mouseRightButtonDownAction = new InvokeCommandAction()
+            {
+                Command = viewModel.NodePortContextMenuCommand,
+                CommandParameter = viewModel
+            };
```

Comment:
Are we missing adding `mouseRightButtonDownTrigger` and `mouseRightButtonDownAction` to the trigger like the others?

Reply:
do you mean?  Which others?

Reply:
I meant there are other triggers like mouseLeftButtonDownTrigger above, which was added like:
```
mouseLeftButtonDownTrigger.Actions.Add(mouseLeftButtonDownAction);
Dynamo.Microsoft.Xaml.Behaviors.Interaction.GetTriggers(MainGrid).Add(mouseLeftButtonDownTrigger);
```
but I don't see mouseRightButtonDownTrigger and mouseRightButtonDownAction being used anywhere